[PLATFORM-1149] Markdown formatting styles

juhah opened the pull request:
Styling of the generated markdown text. Note that these are not finished, just added some default styles as example:

<img width="708" alt="Screen Shot 2019-11-19 at 14 28 22" src="https://user-images.githubusercontent.com/1064982/69126664-4762e300-0adb-11ea-84f8-1b456595f2eb.png">

<img width="577" alt="Screen Shot 2019-11-19 at 14 34 22" src="https://user-images.githubusercontent.com/1064982/69126646-40d46b80-0adb-11ea-8189-3a2b1b6a5845.png">

@mattatgit Can you do a pass on these text stylings?

mattatgit commented:
@juhah ok, will get you something soon.

mattatgit commented:
Actually one thing i think we should do first up - remove H1 & H2 styling, realised that will just be wildly overused & we're better off just limiting people to bold, italics, quote, bullet list, ordered list, code block

mattatgit commented:
Ok, so here's a link to an Abstract collection showing Markdown edit & display views

https://share.goabstract.com/19f64ff3-1324-4ca8-beb4-74cccad8a6c4

This is not in Core Master btw, just in my private WIP file, so a bit quick & dirty, but should do the job for now & helps you to move forward without delay.

And some notes on styles

Generally i think the styles here are ok. But here are some specs:

* Bold - should not be the actual Bold weight in the Plex font, but the Medium weight
* italics - make sure we're using Plex Italics not slanted Regular (looks like you are)
* code style. Here we need to use Mono, looks like you are already. Rather than pink, i'd stick with the same text colour, but put in in a grey box background. So 14pt Mono (rather than the 18pt Sans), with #E7E7E7 box behind it. A 1px radius on box corners will look good.
* for the bullet list, remove the indent and align the bullets with the left edge of the text block
* block quote, i’ll put this into the sketch doc

All these styles are in the Sketch file, just adding them here for convenience
@juhah

juhah commented:
@mattatgit Here's how it looks now:
<img width="675" alt="Screen Shot 2019-11-21 at 10 06 08" src="https://user-images.githubusercontent.com/1064982/69300719-9c6f3800-0c46-11ea-8f85-2c5eb4a85973.png">

> italics - make sure we're using Plex Italics not slanted Regular (looks like you are)

There is no separate font for italics but we are importing the `IBM Plex Sans` `400i` variant and I added the `font-weight: 400` for the `em` tag. 

> for the bullet list, remove the indent and align the bullets with the left edge of the text block

Added also numbered list in the same style.

Unfortunately the PR staging url is not accessible when the CI is broken but let me know if you need more screenshots or there something to fix.

mattatgit commented:
@juhah 

Italics, yeah it's just a weight of Plex, not a separate font altogether, looks like you're using the right one. All looks pretty good to me, the only thing i'd adjust is the spacing between bullets / numbers / quote marker, which looks just a touch wide.

Problem is that we want the text contents of lists, ordered lists and block quotes to all neatly left align with each other, but they all need slightly different spacing.

Here's an example from Dropbox Paper.
They solve it by moving numbers more to the left, and aligning the block quote markers underneath the numbers. I'd be happy with this method.

<img width="1405" alt="Screenshot 2019-11-21 14 23 17" src="https://user-images.githubusercontent.com/6858525/69301740-609a9980-0c6b-11ea-97cc-724ab36a4c8b.png">



So,

* tighten up spacing between bullets and text block by maybe 25%
* move numbers in numbered list slightly left, so periods align under bullets (as Dropbox do)
* maybe block quote marker aligns under numbers, as Dropbox does.

juhah commented:
> move numbers in numbered list slightly left, so periods align under bullets

<img width="706" alt="Screen Shot 2019-11-21 at 11 58 54" src="https://user-images.githubusercontent.com/1064982/69305606-3212c380-0c57-11ea-99d8-a8cc784605a4.png">

@mattatgit This has now slightly narrower gap and the dots align. Numbers from 100 and over won't align correctly but I doubt anyone has that long a list :) But can also make the gap narrower and accommodate larger numbers.

mattatgit commented:
Looks pretty good. I think though, let's just keep bullets, numbers and block quote marker to the same colour as the text, probably #525252 i think, rather than the light grey. Keep it a bit simple. Then i think it's fine.

No worries about the long numbers, if someone puts a list numbered to 100, what are they doing?

juhah commented:
@mattatgit Yep, got it - here's the last pass without separate bullet and quote coloring:

<img width="665" alt="Screen Shot 2019-11-21 at 13 35 10" src="https://user-images.githubusercontent.com/1064982/69313327-d4d13f00-0c63-11ea-9119-2892fe9d8060.png">

mattatgit commented:
Looks good to me
